frontend/public/components/secrets/create-secret/OpaqueSecretForm.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Button } from '@patternfly/react-core';
+import { MinusCircleIcon } from '@patternfly/react-icons/dist/esm/icons/minus-circle-icon';
+import { PlusCircleIcon } from '@patternfly/react-icons/dist/esm/icons/plus-circle-icon';
+import { SecretSubFormProps, OpaqueDataEntry } from './types';
+import { OpaqueSecretFormEntry } from './OpaqueSecretFormEntry';
+import { opaqueSecretObjectToArray, newOpaqueSecretEntry, opaqueEntriesToObject } from './utils';
+import { size, map } from 'lodash';
+
+export const OpaqueSecretForm: React.FC<SecretSubFormProps> = ({ onChange, stringData }) => {
+  const { t } = useTranslation();
+  const [opaqueDataEntries, setOpaqueDataEntries] = React.useState(
+    opaqueSecretObjectToArray(stringData),
+  );
+
+  React.useEffect(() => {
+    onChange({
+      base64StringData: opaqueEntriesToObject(opaqueDataEntries),
+    });
+  }, [opaqueDataEntries, onChange]);
+
+  const onDataChanged = React.useCallback((updatedEntry: OpaqueDataEntry, entryIndex: number) => {
```

Comment:
nit, for consistency
```suggestion
  const updateEntry = React.useCallback((updatedEntry: OpaqueDataEntry, entryIndex: number) => {
```